### ISSUE TITLE
Idea - Audio Library for Word Pronounciations

### ISSUE THREAD

**Issue by rickhumphries** [elmsln/elmsln#620]
This is probably an idea that would be more useful in the sciences, but could have interesting applications elsewhere too.

Text-to-speech does a very poor job with the pronunciation of complex scientific words. In our case, properly pronouncing drug/medicine names is important for our learners to communicate medications to patients, but it's always difficult to decipher how a word should be pronounced from how it's spelt. 

http://howjsay.com/ has an online library of audio clips where the word is read out properly, which is where this idea came from.

Therefore my plan was to record short audio clips reading specific words out, upload them to Media - then have an option within the course (possibly in the accessibility settings) to switch on a 'medicine name audio library' to get the following effect when rolling your mouse over (or clicking) the word in your course: 
http://codepen.io/rickhumphries/pen/MKLYXP
(try typing metronidazole into https://www.ivona.com/ and you'll see how different they can be!)

My thinking is that the instructor could create their own libraries of words - and link each word to an audio shortcode from Media. The student would have the option in the accessibility settings to switch on/off a specific library. If a library is switched on, the tool scans the book for all instances of this word and processes it in a way so that of you click the word, the audio file plays.

Just wondering if this is an interesting concept for anyone else - and to get implementation ideas too...

**Comment by heyMP** [comment]
I love it!  Such a useful feature.  What if we took it in two steps?

- We implement a new content type in icor called "**Pronunciations**".  Once you create a new pronunciation on the icor system you would be able to use the placeholder token/shortcode to place it anywhere in your course content.  The **Pronunciations** content type would have the following fields
 - Name(the name of the word that will be displayed)
 - Audio file upload (for uploading custom audio files)
 - External link (link to an externally hosted audio file such as howjsay.com)
 - Course Reference
 - Tags

- We then implement an automated piece to this process. Possible options:
 - Create a text filter that replaces marked words in the body field with it's token/shortcode replacement from icor
 - Use something destructive like search and replace to replace all words with their icor shortcodes
 - Possibly tie into the glossary?

I'm going to show this to our group and see what they think.  I'm sure we can get this at least partially implemented soon.

**Comment by heyMP** [comment]
Also, this could be a good opportunity to add HTML5 audio recording to ELMS.  In the pronunciations content type, it would be used by admins to optionally record their audio straight from their browser.  In other instances, it could be used as an evaluation tool, such as within a linguistics course.

**Comment by btopro** [comment]
Which we have a widget for. Search drupal.psu.edu for html5 recording widget!

Wp8
________________________________

**Comment by rickhumphries** [comment]
Yeh great plan @heyMP ! And I think the HTML5 audio recording straight from the browser would be fantastic - such a great way to reduce the barriers for instructors to add these in :) :) Let me what your group thinks

**Comment by btopro** [comment]
almost 2 years ago -- https://www.youtube.com/watch?v=kNUSmWpeWsk

**Comment by btopro** [comment]
@rickhumphries love the idea, I was texting @heyMP that he's thinking too much like me if he beat me to the punch on where we could take this idea. Our group may have less direct usage for this but I know our art history and world music courses could benefit from this concept. @kat-wehr any applicability in the business realm?

**Comment by kat-wehr** [comment]
There's definitely an opportunity here for audio recording in conjunction with slides or other media in a presentation style assignment. I'm thinking like VoiceThread except with a nicer interface (and free/open source of course). The application would be HUGE for folks learning second languages, or to help faculty create databases of commonly-used vocabulary for ESL students and offer it as a resource for those learners. Also, giving essay/written test responses orally is a common accommodation for students, I could see a tool like this being used in that capacity as well.

**Comment by btopro** [comment]
@kfurst thoughts from a K-12 perspective? The multi-lingual / ESL support immediately comes to mind, good idea on that @kat-wehr .

**Comment by aitala** [comment]
Adding to @heyMP's implementation, it might be useful to also have an entity reference which could point to related words/pronunciations.  I was thinking of brand vs. generic names of medications or stars & the constellations in which they reside.

**Comment by Chrissuli** [comment]
When thinking of a multi-cultural environment where English maybe for many a second or third language, this is a reasonable idea.

However, thinking about the function / feature from a layman's perspective I am reminded of the famous Winston Churchill quote (in not so many words) that:

" Never were two countries so divided by a common language"

Which language are you proposing to use, English or American!?

Then also consider that an English billion is not the same as an American billion ;)

Kr,
Chris

**Comment by rickhumphries** [comment]
@Chrissuli  Yes, pronunciations can be different - but by creating your own audio then you can at least localise it to an extent. 

Alternatively - you can record a slightly longer audio and try to cover multiple common pronunciations, e.g. http://howjsay.com/mp3/capillary.mp3 if the course has a more global scope.

**Comment by btopro** [comment]
we also wouldn't be packaging audio, just the ability to provide it. multi-track might be good too for a simple multiple pronunciation option; that or for things like saying the word paired with saying the word in a sentence as two separate files.

**Comment by Chrissuli** [comment]
Multiples would be needed for some words and syllables that have differing pronunciations for the same spelling depending on the context, e.g. noun vs. verb variations.
I agree the sound track quality does not need to be high fidelity.
Another though that comes to mind when thinking of the larger purpose embodied in this idea is that done by some modern web services in which key words in a text are automatically identified and linked to relevant information, such as a Wikipedia page or thesaurus.
If this can be done in this language context or perhaps even combined in one system of dynamic content linking might be useful. I suspect a manual override might be needed though as otherwise this may cause a hyper-link overload of less relevant info and distract from links to info more directly relevant to the lesson. 

/ Chris

**Comment by rickhumphries** [comment]
http://soundcite.knightlab.com

**Comment by btopro** [comment]
woah. that's awesome!
